src/deploymentmanager.ts

```diff
@@ -163,6 +167,34 @@ export class DeploymentManager implements IDeploymentManager {
                 deployUI.start('', options);
                 return this._client.deployments.createOrUpdate(answers.solutionName as string, deploymentName, deployment);
             })
+            .then((res: DeploymentExtended) => {
+                if (this._solutionType === 'remotemonitoring' && res.properties) {
+                    const keyVaultDeploymentName = deploymentName + '-keyvault';
+                    const armTemplatePath = __dirname + path.sep + 'solutions' + path.sep + this._solutionType + path.sep + 'armtemplates' + path.sep;
+
+                    try {
+                        this.setupKeyvaultParameters(answers, res.properties.outputs);
+                    } catch (ex) {
+                        throw new Error('Could not find template or parameters file for KeyVault, Exception:');
+                    }
+
+                    const keyVaultTemplate = require(armTemplatePath + 'keyvault.json');
+                    const keyVaultDeployment: Deployment = {
+                        properties: {
+                            mode: 'Incremental',
+                            parameters: this._keyVaultParams,
+                            template: keyVaultTemplate
+                        }
+                    };
+
+                    return this._client.deployments.createOrUpdate(answers.solutionName as string, keyVaultDeploymentName, keyVaultDeployment)
```

Comment:
Do you still need (this)[https://github.com/Azure/pcs-cli/pull/476/files#diff-a25cfaebe1be93b318c70debfe07a223R213]? Are you just keeping it in for now and remove later?

Reply:
I continued to follow the pattern because it actually came in handy in an unexpected way. It helped me provide a list of all the parameters in a concise view while being able to see what had / hadn't been filled in. With that said now that it is done it may not provide as much utility. It can be debated either way though as whether or not we should continue to resemble what is done by the other ARM templates. Thoughts?

Reply:
I might put the comment in the wrong place. What you are doing here makes sense since you are running another deployment to setup KV.